packages/vscode-extension/src/devices/IosSimulatorDevice.ts

```diff
@@ -263,8 +263,43 @@ export class IosSimulatorDevice extends DeviceBase {
     }
   }
 
+  /**
+   * This function terminates any running applications. Might be useful when you launch a new application
+   * before terminating the previous one.
+   */
+  async terminateAnyRunningApplications() {
+    const deviceSetLocation = getOrCreateDeviceSet(this.deviceUDID);
+    const { stdout } = await exec("xcrun", [
+      "simctl",
+      "--set",
+      deviceSetLocation,
+      "listapps",
+      this.deviceUDID,
+    ]);
+
+    const regex = /ApplicationType = User;\s*[^{}]*?\bCFBundleIdentifier = "([^"]+)/g;
+
+    let match;
+    while ((match = regex.exec(stdout)) !== null) {
+      const bundleID = match[1];
+      // Terminate the app if it's running:
+      try {
+        await exec(
+          "xcrun",
+          ["simctl", "--set", deviceSetLocation, "terminate", this.deviceUDID, bundleID],
+          { allowNonZeroExit: true }
+        );
```

Comment:
would likely be better to use Promise.all to await on all the commands

Reply:
done